### PR TITLE
Remove light filtering data

### PR DESCRIPTION
**Pull request by EatThePath**
`light_ignore_objnum` and `affected_objnum` support features that, as far as I can determine, can't work in deferred and don't even work in non-deferred lighting. Their continued presence in the code is a distraction.

Submitted in draft at the moment because my ability to actually test these changes has been very limited.

### CODE REVIEW

**Comment by EatThePath** [comment]
The commit log mess is from me rebasing on current master . I can withdraw this PR and resubmit the changes in a new branch if that's going to make a mess.

**Comment by EatThePath** [comment]
@wookieejedi It's hard to prove a negative, which is the situation here as I don't see how the removed code would have any visible impact on the game. However I've run the game in lighting heavy situations both with and without the change, and with and without -no_deferred, and I can see no difference to my eye.

**Review by asarium — APPROVED** [review]
I agree that such a feature is not possible to implement in our current lighting setup so there is no reason to keep this code.

**Review by z64555 — CHANGES_REQUESTED** [review]
Looks good otherwise for a few style/formatting nitpicks

**Review by z64555 — APPROVED** [review]

**Comment by z64555** [comment]
Final call for testing before merge.

**Comment by EatThePath** [comment]
I've been working this weekend on a branch incorporating these changes and they're working fine, so I'm pushing the button

**Pull request merged**
